crates/rpc/src/v02/method/get_state_update.rs

```diff
@@ -60,56 +60,36 @@ pub async fn get_state_update(
 fn block_info(
     tx: &rusqlite::Transaction<'_>,
     block: StarknetBlocksBlockId,
-) -> anyhow::Result<Option<(StarknetBlockNumber, StarknetBlockHash, StateCommitment)>> {
-    let output = match block {
-        StarknetBlocksBlockId::Number(number) => {
-            let output = tx.query_row(
-                r"SELECT starknet_blocks.hash, starknet_blocks.root FROM starknet_blocks
-                    JOIN canonical_blocks ON (starknet_blocks.hash = canonical_blocks.hash)
-                    WHERE starknet_blocks.number = ?", [number], |row| {
-                        let hash: StarknetBlockHash = row.get(0)?;
-                        let root: StateCommitment = row.get(1)?;
-                        Ok((hash, root))
-                    })
-                    .optional()
-                    .context("Querying block info by number")?
-                    .map(|(hash, root)| (number, hash, root));
-                Ok(output)
-        },
-        StarknetBlocksBlockId::Hash(hash) => {
-            let output = tx.query_row(
-                r"SELECT starknet_blocks.number, starknet_blocks.root FROM starknet_blocks
-                    JOIN canonical_blocks ON (starknet_blocks.hash = canonical_blocks.hash)
-                    WHERE starknet_blocks.hash = ?", [hash], |row| {
-                        let number: StarknetBlockNumber = row.get(0)?;
-                        let root: StateCommitment = row.get(1)?;
-                        Ok((number, root))
-                    })
-                    .optional()
-                    .context("Querying block info by hash")?
-                    .map(|(number, root)| (number, hash, root));
-                Ok(output)
-            },
-        StarknetBlocksBlockId::Latest => tx.query_row(
-            r"SELECT starknet_blocks.number, starknet_blocks.hash, starknet_blocks.root FROM starknet_blocks
-                JOIN canonical_blocks ON (starknet_blocks.hash = canonical_blocks.hash)
-                ORDER BY starknet_blocks.number DESC LIMIT 1",
-            [],
-            |row| Ok((row.get(0)?, row.get(1)?, row.get(2)?)),
-        ).optional().context("Querying latest block info"),
+) -> Result<
+    (
+        StarknetBlockNumber,
+        StarknetBlockHash,
+        StateCommitment,
+        StateCommitment,
+    ),
+    GetStateUpdateError,
+> {
+    let block = StarknetBlocksTable::get(tx, block)?.ok_or(GetStateUpdateError::BlockNotFound)?;
+
+    let old_root = if block.number == StarknetBlockNumber::GENESIS {
+        StateCommitment(Felt::ZERO)
+    } else {
+        let previous_block_number = StarknetBlockNumber::new_or_panic(block.number.get() - 1);
+        let previous_block =
+            StarknetBlocksTable::get(tx, StarknetBlocksBlockId::Number(previous_block_number))?
```

Comment:
No, I don't think we would win too much. The problem is that querying blocks is a bit complicated from the DB (because `root` is _not_ the state commitment, only the storage commitment) so I'd like to keep the query / state commitment computation logic in one place.

I don't think we lose much (compared to all other queries) by having to decode those extra few columns from the _same_ row.

Reply:
👍Yeah I really wish we had just stored it instead of this extra calculation step. Might become quite costly with serving p2p.